### DISCUSSION TITLE
Documentation confusing - push/new screen by ID #

### DISCUSSION THREAD

**Discussion by ehenighan** [Instawork/hyperview#650] [Q&A]
Hi,

The reference for the 'push' and 'new' navigation actions (https://hyperview.org/docs/reference_behavior_attributes#new) says 'the fetched content must be either a full Hyperview XML document (with <doc> root), or a <screen> element from the current screen's document'

'a <screen> element from the current screen's document' seems like it must mean the `href="#screen-id"` pattern, but the reference for the 'href' attribute suggests that this isn't supported in the current version as only update actions are currently supported, not navigation actions: https://hyperview.org/docs/reference_behavior_attributes#document-fragments

I think, based on this, to make an overlaid navigation modal-style menu I'd need to use a button with multiple show/hide behaviours instead?

Could anyone clarify what the current state is, and if there's any better option for an overlaid modal menu?

Thanks,

Ed

**Comment by ehenighan** [comment]
To clarify - to make an overlaid modal menu without an additional server round-trip. It looks like it should be possible by just rendering the menu 'screen' into the main document on initial render alongside the main display screen and then pushing it onto the screen stack/toggling visibility as needed, but I'm struggling to make it work

**Comment by adamstep** [comment]
Hi @ehenighan , sorry for the confusion in the documentation. I agree it's confusing at the moment. Yes, in the past there was support for multiple `<screen>` s in the same document, and navigation actions like `new` could reference those screen IDs to save a server round-trip.

However, in practice we found this pattern created lots of issues, in particular it was hard to have unique IDs across screens, which would lead to unexpected behavior. We are in the process of phasing out support for this functionality to keep things simpler.

That means for your use case, there are two options:
- make a server round-trip to load the second modal
- use a custom component to show a "bottom-sheet" embedded in the document. This is what we do at Instawork. We could share our implementation (just a single JS file you drop into your project and register) if that could handle your use case.

**Reply by ehenighan** [reply]
Hi Adam,

Sorry for the slow reply - that helps though! I know what my options are now :)

Thanks,

Ed
